### ISSUE TITLE
RangeError: Invalid string length at onChunkRead

### ISSUE THREAD

**Issue by mourawaldson** [GoogleChrome/lighthouse#1685]
**Lighthouse Version**: 1.5.0
**Chrome Version**: 56.0.2924.87
**URL**: https://app.ft.com/index_page/home
**Error Message**: scripts/popup.js:153 RangeError: Invalid string length at onChunkRead (lighthouse-background.js:9579)
background.runLighthouseInExtension.catch.err @ scripts/popup.js:153
**Log**:
```
lighthouse-background.js:36680 status Disconnecting from browser... +6s
scripts/popup.js:153 RangeError: Invalid string length
    at onChunkRead (lighthouse-background.js:9579)
background.runLighthouseInExtension.catch.err @ scripts/popup.js:153
lighthouse-background.js:36680 <= Network.requestWillBeSent +6ms Object {documentURL: "https://app.ft.com/index_page/home", frameId: "18752.18", initiator: Object, loaderId: "18752.23", request: Object…}
```

Note: Since the log is too big I've only added the final lines, let me know if the full log is needed. Also, this was taken from the console log since the app just stopped running without raising any issue to the client level (which in my opinion should be done if any error occur).

**Comment by brendankenny** [comment]
This is interesting. Happening from the trace being streamed back over the protocol.

I'm not able to reproduce, but the page load is timing out for me, so the trace will be fairly large. Maybe it's some kind OOM issue?

**Comment by paulirish** [comment]
Okay fun one. @caseq and I looked into this.

I'm pretty sure it's coming from `  result += response.data;` in onChunkRead.

It's coming from V8 when a string gets to be too big.
What is the max size of a string in V8? 
Glad you asked. The answer is 2^28 bytes.. which is 256 megs.

---

Now you may think, maybe we should build this string in a fashion more efficient than `+=`? Good idea, bucko, and while that _may_ decrease the memory usage, it won't end up with a smaller string length and we'll still hit this bug.

So, if we want to handle a trace this big, we will need to parse it incrementally.

DevTools already has code to do that and we've already loaded all those modules so it shouldn't be too hard. :)

see `WebInspector.TimelineLoader` and `WebInspector.TextUtils.BalancedJSONTokenizer`.

**Comment by mourawaldson** [comment]
@paulirish and @brendankenny,

When trying to reproduce now I get this:
```
status Evaluating: Estimated Input Latency +18s
lighthouse-background.js:13432 Invalid ProtoExpectation: ignored(6900 6900 undefined) File a bug with this trace!
createInteractionRecord @ lighthouse-background.js:13432
(anonymous) @ lighthouse-background.js:13412
findInputExpectations @ lighthouse-background.js:13412
findUserExpectations @ lighthouse-background.js:13436
buildUserModel @ lighthouse-background.js:13436
buildUserModel @ lighthouse-background.js:13424
run @ lighthouse-background.js:13342
Task.RunSynchronously @ lighthouse-background.js:13342
importTraces @ lighthouse-background.js:13424
init @ lighthouse-background.js:11942
compute_ @ lighthouse-background.js:4704
Promise.resolve.then._ @ lighthouse-background.js:4201
```

Then after few seconds the extension crashes.

@brendankenny they way I do is basically this:
1. Open the URL reported.
2. In another tab go to chrome://extensions then click on "background page (Inactive)" link from Lighthouse (Developer Tools window will open up).
3. Go back to the tab from step 1. then "Generate report" on the extension.

At this point you should see the trace on "Developer Tools" window.

**Comment by paulirish** [comment]
@mourawaldson gotcha. that's a separate bug, though im surprised that actually throws. thanks for reporting it regardless..

**Comment by brendankenny** [comment]
fixed by #2593, github just didn't pick up the auto-close
